src/devices/device_models/hydro_generation.jl

```diff
@@ -394,24 +394,27 @@ function budget_constraints!(canonical::Canonical,
 end
 
 function device_budget_param_ub(canonical::Canonical,
-                            budget_data::Vector{Tuple{String, Int64, Float64, Float64}},
+                            budget_data::Vector{Tuple{String, Int64, Float64, Vector{Float64}}},
                             cons_name::Symbol,
                             param_reference::UpdateRef,
                             var_name::Symbol)
 
     time_steps = model_time_steps(canonical)
     variable = get_variable(canonical, var_name)
     set_name = (r[1] for r in budget_data)
-    constraint = _add_cons_container!(canonical, cons_name, set_name, 1) 
-    param = _add_param_container!(canonical, param_reference, names, 1)
+    no_of_budgets = length(budget_data[1][4])
+    time_lengths = time_steps/length(budget_data[1][4])
+    time_chucks = reshape(time_steps, (time_lengths, no_of_budgets))
```

Comment:
Time_chuck?

Reply:
It's just the grouped time steps, so each column for the matrix is one group. Probably should have a better name

Reply:
Sorry typo, its was suppose to be chunk